test/app.test.js

```diff
@@ -67,3 +67,41 @@ test("optional view", done => {
     }
   })
 })
+
+function memoize( fn ) {
+  return function () {
+      console.log(arguments);
+      var args = Array.prototype.slice.call(arguments),
+          hash = "",
+          i = args.length;
+      currentArg = null;
+      while (i--) {
+          currentArg = args[i];
+          hash += (currentArg === Object(currentArg)) ?
+          JSON.stringify(currentArg) : currentArg;
+          fn.memoize || (fn.memoize = {});
+      }
+      return (hash in fn.memoize) ? fn.memoize[hash] :
+      fn.memoize[hash] = fn.apply(this, args);
+  };
+}
+
+test("skips patching memoized vnodes", done => {
```

Comment:
Does this actually check that the same VNode is returned?

Reply:
trying to figure out how to write the test for that...

Reply:
I think another approach to this would be to create a simple demo app with one memoized component and one component not. Then somehow benchmark it, and once we understand the problem well, take another shot at this?

Reply:
You can call the memoized function twice and check for identity equality.